src/Auth0.php

```diff
@@ -245,8 +245,11 @@ public function exchange(): bool
         $user = $this->state->getIdTokenDecoded();
 
         if ($user === null || $this->configuration->getQueryUserInfo() === true) {
-            $user = $this->authentication()->userInfo($this->state->getAccessToken());
-            $user = HttpResponse::decodeContent($user);
+            $response = $this->authentication()->userInfo($response['access_token']);
+
+            if (HttpResponse::wasSuccessful($response)) {
+                $user = HttpResponse::decodeContent($response);
+            }
```

Comment:
Unrelated to this PR really, but had forgotten to do a success check here before attempting to decode the content, so sneaking that in here.